src/node_url.cc

```diff
@@ -46,18 +45,20 @@ enum url_update_action {
   kHref = 9,
 };
 
-void SetArgs(Environment* env, Local<Value> argv[10], const ada::result& url) {
+void SetArgs(Environment* env,
+             Local<Value> (*argv)[10],
+             const ada::result& url) {
```

Comment:
Ideally this parameter should be an `const ada::url&`. It doesn't make sense to pass an `ada::result` in when we always expect it to be filled with the "valid" value.

In fact the call sites should probably change from this:

```c++
  ada::result out = ada::parse(input.ToStringView());
  CHECK(out);
  out->set_protocol(…);
```

to

```c++
  ada::result out = ada::parse(input.ToStringView());
  CHECK(out);
  const ada::url& url = out.value();
  url.set_protocol(…);
```

Reply:
`out.value()` returns a copy. The current implementation looks unsafe, but `CHECK(out)` ensures that it is not. I tried to avoid returning a copy. @lemire wrote about this on Ada's discussion board: https://github.com/ada-url/ada/discussions/200

Reply:
I think you misunderstood https://github.com/ada-url/ada/discussions/200. `out.value()` (and equivalently `*out`) does not return a copy by itself. It returns a reference, so if you put it into a `const ada::url&` then no copies are made.

On the other hand, `ada::url url = *out`, which *would* make a copy. That's not what I'm suggesting here.

Alternatively, you can also do

```c++
  ada::result out = ada::parse(input.ToStringView());
  CHECK(out);
  ada::url url = std::move(*out);
```

which is written under the "performance tip". It would also avoid a copy.

Reply:
It is likely that the current code is safe and efficient. I think we are discussing 'coding style' which is subjective.

I don't personally find the `url->...` annoying. 

I am slightly triggered by the `(*argv)[..]` however. :-)

What about using a reference to an array instead?

```
Local<Value> (&argv)[10]
```

(The ampersand would be dropped from the calling site.)

Reply:
It's a style guide thing: pointers for things that are mutated, const references otherwise.

With mutable references it's sometimes ambiguous to a reader if code operates on the original or on a copy. With pointers, no such ambiguity exists.